stdlib/mexpr/pprint.mc

```diff
@@ -150,16 +154,41 @@ lang FunPrettyPrint = FunAst
 end
 
 lang RecordPrettyPrint = RecordAst
+  sem record2tuple =
+  | TmRecord t ->
+    match t.bindings with [] then
+      None ()
+    else
+      let keys = assocKeys {eq=eqstr} t.bindings in
+      match all stringIsInt keys with false then None () else
+      let intKeys = map string2int keys in
+      let sortedKeys = sort subi intKeys in
+      -- Check if keys are a sequence 0..(n-1)
+      match and (eqi 0 (head sortedKeys))
+                (eqi (subi (length intKeys) 1) (last sortedKeys)) with true then
+        Some (map (lam key. assocLookupOrElse {eq=eqstr} (lam _. never)
+                                              (int2string key) t.bindings)
+                   sortedKeys)
```

Comment:
These lookups give quadratic complexity, while sorting the association map directly would give nlog(n), however, that would require `assocToList` or `assocSort` or similar, which I'm not sure that we want.

Reply:
Given the most likely small sizes involved here I don't think it matters, but maybe worth leaving a comment noting the optimization opportunity, should it become relevant down the line?